src/category_theory/equivalence.lean

```diff
@@ -1,156 +1,256 @@
+by { erw [←iso.comp_hom_eq_id (e.counit_iso.app _), functor_unit_comp], refl }
+
+lemma counit_functor (e : C ≌ D) (X : C) :
+  e.counit.app (e.functor.obj X) = e.functor.map (e.unit_inv.app X) :=
+by { erw [←iso.hom_comp_eq_id (e.functor.map_iso (e.unit_iso.app X)), functor_unit_comp], refl }
+
+/-- The other triangle equality. The proof follows the following proof in Globular:
+  http://globular.science/1905.001 -/
+lemma unit_inverse_comp (e : C ≌ D) (Y : D) :
+  e.unit.app (e.inverse.obj Y) ≫ e.inverse.map (e.counit.app Y) = 𝟙 (e.inverse.obj Y) :=
 begin
-  erw [nat_iso.naturality_2],
-  refl
+  rw [←id_comp _ (e.inverse.map _), ←map_id e.inverse, ←counit_inv_functor_comp, map_comp,
+      ←iso.hom_inv_id_assoc (e.unit_iso.app _) (e.inverse.map (e.functor.map _)),
+      app_hom, app_inv, unit_def, unit_inv_def],
+  slice_lhs 2 3 { erw [e.unit.naturality] },
+  slice_lhs 1 2 { erw [e.unit.naturality] },
+  slice_lhs 4 4
+  { rw [←iso.hom_inv_id_assoc (e.inverse.map_iso (e.counit_iso.app _)) (e.unit_inv.app _)] },
+  slice_lhs 3 4 { erw [←map_comp e.inverse, e.counit.naturality],
+    erw [(e.counit_iso.app _).hom_inv_id, map_id] }, erw [id_comp],
+  slice_lhs 2 3 { erw [←map_comp e.inverse, e.counit_iso.inv.naturality, map_comp] },
+  slice_lhs 3 4 { erw [e.unit_inv.naturality] },
+  slice_lhs 4 5 { erw [←map_comp (e.functor ⋙ e.inverse), (e.unit_iso.app _).hom_inv_id, map_id] },
+  erw [id_comp],
+  slice_lhs 3 4 { erw [←e.unit_inv.naturality] },
+  slice_lhs 2 3 { erw [←map_comp e.inverse, ←e.counit_iso.inv.naturality,
+    (e.counit_iso.app _).hom_inv_id, map_id] }, erw [id_comp, (e.unit_iso.app _).hom_inv_id], refl
 end
+
+lemma inverse_counit_inv_comp (e : C ≌ D) (Y : D) :
+  e.inverse.map (e.counit_inv.app Y) ≫ e.unit_inv.app (e.inverse.obj Y) = 𝟙 (e.inverse.obj Y) :=
+begin
+  erw [iso.inv_eq_inv
+    (e.unit_iso.app (e.inverse.obj Y) ≪≫ e.inverse.map_iso (e.counit_iso.app Y)) (iso.refl _)],
+  exact e.unit_inverse_comp Y
+end
+
+lemma unit_inverse (e : C ≌ D) (Y : D) :
+  e.unit.app (e.inverse.obj Y) = e.inverse.map (e.counit_inv.app Y) :=
+by { erw [←iso.comp_hom_eq_id (e.inverse.map_iso (e.counit_iso.app Y)), unit_inverse_comp], refl }
+
+lemma inverse_counit (e : C ≌ D) (Y : D) :
+  e.inverse.map (e.counit.app Y) = e.unit_inv.app (e.inverse.obj Y) :=
+by { erw [←iso.hom_comp_eq_id (e.unit_iso.app _), unit_inverse_comp], refl }
+
+@[simp] lemma fun_inv_map (e : C ≌ D) (X Y : D) (f : X ⟶ Y) :
+  e.functor.map (e.inverse.map f) = e.counit.app X ≫ f ≫ e.counit_inv.app Y :=
+(nat_iso.naturality_2 (e.counit_iso) f).symm
+
 @[simp] lemma inv_fun_map (e : C ≌ D) (X Y : C) (f : X ⟶ Y) :
-e.inverse.map (e.functor.map f) = (e.fun_inv_id.hom.app X) ≫ f ≫ (e.fun_inv_id.inv.app Y) :=
+  e.inverse.map (e.functor.map f) = e.unit_inv.app X ≫ f ≫ e.unit.app Y :=
+(nat_iso.naturality_1 (e.unit_iso) f).symm
+
+section
+open category_theory
+variables {F : C ⥤ D} {G : D ⥤ C} (η : 𝟭 C ≅ F ⋙ G) (ε : G ⋙ F ≅ 𝟭 D)
```

Comment:
As Lean forces the auxiliary definitions to go first, it might help to add a one-line explanatory comment somewhere around here, like `-- Convert an arbitrary equivalence to a half-adjoint equivalence.`